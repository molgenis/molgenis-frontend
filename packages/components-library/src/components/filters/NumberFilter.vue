<template>
  <div>
    <b-input-group>
      <b-form-input
        v-model="model"
        type="number"
        :min="min"
        :max="max"
        :step="step"
      />
    </b-input-group>
    <vue-slider
      v-if="useSlider"
      v-model="model"
      :min="min"
      :max="max"
      :interval="step"
      class="mt-2"
    />
  </div>
</template>

<script>
import Vue from 'vue'
import VueSlider from 'vue-slider-component'
import 'vue-slider-component/theme/default.css'

export default Vue.extend({
  name: 'NumberFilter',
  components: { VueSlider },
  props: {
    name: {
      type: String,
      required: true
    },
    min: {
      type: Number,
      default: () => Number.MIN_SAFE_INTEGER
    },
    max: {
      type: Number,
      default: () => Number.MAX_SAFE_INTEGER
    },
    step: {
      type: Number,
      default: () => 1
    },
    value: {
      type: Number,
      default: () => null
    },
    useSlider: {
      type: Boolean,
      default: () => false
    }
  },
  computed: {
    model: {
      get () {
        return this.value
      },
      set (value) {
        if (typeof value === 'string' && value === '') {
          this.$emit('input', undefined)
        } else {
          value = Math.min(Math.max(value, this.min), this.max)
          this.$emit('input', parseFloat(value))
        }
      }
    }
  }
})
</script>
