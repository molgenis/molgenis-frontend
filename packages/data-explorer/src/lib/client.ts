import build from './clientFactory'
import store from '@/store/store'
import { AxiosInstance } from 'axios'

export const errorReponse = (error: any) => {
  let message = error.message
  if (error.response && error.response.data && error.response.data.detail) {
    message = error.response.data.detail
  }
  if (error.response && error.response.status === 401) {
    if (!store.getters.isUserAuthenticated) {
      window.location.href = '/login'
    }
  }
  // Set timeout to '0' to keep visible until user dismisses the error notification
  store.commit('addToast', { message, type: 'danger', timeout: 0 })
  return Promise.reject(error)
}

// Create the default axios client the data-explorer will use
const client:AxiosInstance = build({
  responseErrorInterceptor: errorReponse
})

export default client
