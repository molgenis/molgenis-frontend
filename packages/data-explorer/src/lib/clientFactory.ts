/* eslint-disable no-console */
import axios, { AxiosInstance, AxiosResponse } from 'axios'
import { ClientSettings } from '@/types/ClientSettings'

export const defaultResponse = function (response: AxiosResponse<any>) {
  return response
}

export const defaultErrorReponse = function (error: any) {
  if (console && console.error) {
    console.error(`${error.errorCode}: ${error.title}`)
    console.error(`${error.detail}`)
  }
  return Promise.reject(error)
}

export const defaultStatusValidation = function (status: number) {
  return status >= 200 && status < 300
}

export function build (settings?: ClientSettings):AxiosInstance {
  const client:AxiosInstance = axios.create({
    baseURL: settings && settings.baseURL ? settings.baseURL : '/',
    timeout: settings && settings.timeout ? settings.timeout : 5000,
    validateStatus: settings && settings.validateStatus ? settings.validateStatus : defaultStatusValidation
  })

  client.interceptors.response.use(
    settings && settings.responseInterceptor ? settings.responseInterceptor : defaultResponse,
    settings && settings.responseErrorInterceptor ? settings.responseErrorInterceptor : defaultErrorReponse
  )

  return client
}

export default build
