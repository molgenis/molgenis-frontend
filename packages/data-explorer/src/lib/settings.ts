import Axios from 'axios'
import store from '../store/store'

function getSettingsUrl () {
  return `${window.location.href.replace(window.location.hash, '').replace('index.html', '')}config/settings.xlsx`
}

function setSettingsFormData (fileContents: string) {
  let formData = new FormData()
  let dataAsBlob = new Blob([fileContents], { type: 'application/xml' })
  formData.append('file', dataAsBlob, 'settings.xlsx')
  formData.append('action', 'ADD_UPDATE_EXISTING')
  formData.append('notify', 'false')
  return formData
}

function settingsByUrlFormdata () {
  return `/plugin/importwizard/importByUrl?notify=true&entityTypeId=demo&url=${encodeURIComponent(getSettingsUrl())}`
}

const BootstrapExplorer = async () => {
  let versionNumberRequest = await Axios.get('/config/version.txt')
  const version = versionNumberRequest.data
  // check for table
  // if not then check user rights
  // const fileRequest = await Axios.get('/config/settings.xlsx')
  // const settingsFormData = setSettingsFormData(fileRequest.data)
  // console.log(fileRequest)
  const t = settingsByUrlFormdata()
  let x = await Axios.post(t)

  // /plugin/importwizard/importByUrl?notify=false&entityTypeId=demo&url=http://localhost:8080/config/settings.xlsx
  if (store.getters.hasEditRights) { }

  // Axios.get('/config/settingstable.xlsx').then(t => console.log(t))
  let isSettingsTableAvailable = Axios.get('/api/metadata/root_hospital_patients?flattenAttributes=true').then(t =>
    console.log(t)
  )
}

export default BootstrapExplorer
