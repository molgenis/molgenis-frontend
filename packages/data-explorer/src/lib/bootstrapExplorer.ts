import client from './client'
import store from '../store/store'
import applicationSettings from './applicationSettings'

const packageEndpoint = 'api/data/sys_md_Package'
const metaDataEndpoint = 'api/metadata'

async function createSettings () {
  if (!store.getters.hasEditRights) {
    return {
      type: 'danger',
      message: 'Please login as administrator to initialize the application'
    }
  }

  try {
    await client.post(packageEndpoint, applicationSettings.packageSettings)
  } catch (error) {
    // if the settings table already exists, axios gives error and stops executing.
  }

  await client.post(metaDataEndpoint, applicationSettings.entitySettings)
  return {
    type: 'success',
    message: 'The application has been succesfully initialized!'
  }
}

const bootstrapExplorer = async (): Promise<any | undefined> => {
  try {
    await client.get('/api/data/app_set_DataExplorerEntitySettings')
  } catch (error) {
    store.commit('setToast', { type: 'warning', message: 'Bootstrapping Data explorer' })
    const status = await createSettings()
    store.commit('setToast', { type: status.type, message: status.message })
  }
}

export default bootstrapExplorer
