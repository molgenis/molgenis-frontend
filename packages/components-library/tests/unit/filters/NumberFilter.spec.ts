import { mount } from '@vue/test-utils'
import NumberFilter from '@/components/filters/NumberFilter.vue'
import { localVue } from '../../lib/helpers'

describe('NumberFilter.vue', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = mount(NumberFilter, {
      localVue: localVue(),
      propsData: {
        name: 'name',
        label: 'label',
        value: 42
      }
    })
  })

  it('sets value property on inner input', () => {
    const inputElement = wrapper.find('input').element as HTMLInputElement
    expect(inputElement.value).toBe('42')
  })

  it('triggers event after value change', () => {
    wrapper.find('input').setValue(20)
    wrapper.find('input').trigger('change')
    expect(wrapper.emitted()).toEqual({ input: [[20]] })
  })

  it('emits undefined when value is set empty string ', () => {
    wrapper.find('input').setValue('')
    wrapper.find('input').trigger('change')
    expect(wrapper.emitted()).toEqual({ input: [[undefined]] })
  })
})
