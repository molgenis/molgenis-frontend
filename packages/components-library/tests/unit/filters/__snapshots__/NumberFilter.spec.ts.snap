// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberFilter.vue matches the snapshot 1`] = `
<div
  label="label"
  name="name"
>
  <div
    class="input-group"
    role="group"
  >
    <!---->
    <input
      class="form-control"
      id="__BVID__4"
      max="9007199254740991"
      min="-9007199254740991"
      step="1"
      type="number"
    />
    <!---->
  </div>
   
  <!---->
</div>
`;
